FrontDesk/src/FrontDesk.Blazor/Pages/ClientsPage.razor

```diff
@@ -0,0 +1,199 @@
+                    <td>
+                        @if (IsAddOrEdit(@client.ClientId))
+                        {
+                            <input @bind="ToSave.FullName" />
+                        }
+                        else
+                        {
+                            @client.FullName
+                        }
+                    </td>
+                    <td>
+                        @if (IsAddOrEdit(@client.ClientId))
+                        {
+                            <input @bind="ToSave.EmailAddress" />
+                        }
+                        else
+                        {
+                            @client.EmailAddress
+                        }
+                    </td>
+                    <td>
+                        @if (IsAddOrEdit(@client.ClientId))
+                        {
+                            <input @bind="ToSave.Salutation" />
+                        }
+                        else
+                        {
+                            @client.Salutation
+                        }
+                    </td>
+                    <td>
+                        @if (IsAddOrEdit(@client.ClientId))
+                        {
+                            <input @bind="ToSave.PreferredName" />
+                        }
+                        else
+                        {
+                            @client.PreferredName
+                        }
+                    </td>
+                    <td class="action-buttons">
+                        @if (IsAddOrEdit(@client.ClientId))
+                        {
+                            <button @onclick="@(() => SaveClick())" @onclick:stopPropagation="true" class="btn btn-primary margin-5">
+
+                                Save
+                            </button>
+                            <button @onclick="@(() => CancelClick())" @onclick:stopPropagation="true" class="btn btn-danger margin-5">
+
+                                Cancel
+                            </button>
+                        }
+                        else
+                        {
+                            <button @onclick="@(() => EditClick(client.ClientId))" @onclick:stopPropagation="true" class="btn btn-primary margin-5">
+                                <svg width="1em" height="1em" viewBox="0 0 16 16" class="bi bi-pencil-square" fill="currentColor" xmlns="http://www.w3.org/2000/svg">
+                                    <path d="M15.502 1.94a.5.5 0 0 1 0 .706L14.459 3.69l-2-2L13.502.646a.5.5 0 0 1 .707 0l1.293 1.293zm-1.75 2.456l-2-2L4.939 9.21a.5.5 0 0 0-.121.196l-.805 2.414a.25.25 0 0 0 .316.316l2.414-.805a.5.5 0 0 0 .196-.12l6.813-6.814z" />
+                                    <path fill-rule="evenodd" d="M1 13.5A1.5 1.5 0 0 0 2.5 15h11a1.5 1.5 0 0 0 1.5-1.5v-6a.5.5 0 0 0-1 0v6a.5.5 0 0 1-.5.5h-11a.5.5 0 0 1-.5-.5v-11a.5.5 0 0 1 .5-.5H9a.5.5 0 0 0 0-1H2.5A1.5 1.5 0 0 0 1 2.5v11z" />
+                                </svg>
```

Comment:
We need these path elements? That looks really low level. What's wrong with just a button? Or a separate control so this isn't in the form?

Reply:
That was used for icon.
I made components for edit and delete.